lib/Dialect/Torch/Transforms/ShapeLibrary.cpp

```diff
@@ -4506,90 +4502,76 @@ module {
   }
   func.func @__torch__.torch.jit._shape_functions.sum_mean_dim(%arg0: !torch.list<int>, %arg1: !torch.optional<list<int>>, %arg2: !torch.bool, %arg3: !torch.any) -> !torch.list<int> {
     %str = torch.constant.str "AssertionError: "
+    %int0 = torch.constant.int 0
     %false = torch.constant.bool false
     %true = torch.constant.bool true
     %none = torch.constant.none
-    %int0 = torch.constant.int 0
     %int1 = torch.constant.int 1
     %0 = torch.prim.ListConstruct  : () -> !torch.list<int>
     %1 = torch.aten.__is__ %arg1, %none : !torch.optional<list<int>>, !torch.none -> !torch.bool
-    %2 = torch.prim.If %1 -> (!torch.bool) {
```

Comment:
This is undoing changes from my merged PR https://github.com/llvm/torch-mlir/pull/1217. Are you using the latest pytorch nightly to generate this file? The new version of `sum_mean_dim` performs this operation

https://github.com/pytorch/pytorch/blob/e9dff858c3d9aa57d4ecca4410bfbcd996eaf8eb/torch/jit/_shape_functions.py#L166-L167

Reply:
Yes, I am using the latest PyTorch head. It contains your change to sum_mean_dim. That is strange that it would revert this change.